### PR TITLE
Preview player hand

### PR DESCRIPTION
**Pull request by pauchan**
I was trying to show preview of a given player hand and tried to use `show` for that goal. But with the code below I'm getting the following error:
```
/Users/pawel-rusin/Devel/haskell/hasbatteline/app/Main.hs:30:3: error:                                                │data Color = Blue | Green | Orange | Purple | Red | Yellow deriving (Show, Enum)                              
    • Couldn't match type ‘[]’ with ‘IO’                                                                              │data Card = Card Color Int deriving (Show)                                                                    
      Expected type: IO Char                                                                                          │data Flag = One | Two | Three | Four | Five | Six | Seven | Eight | Nine deriving Enum                        
        Actual type: String                                                                                           │data PlayerNumber = Player1 | Player2                                                                         
    • In a stmt of a 'do' block: show (hand player)                                                                   │                                                                                                              
      In the expression:                                                                                              │shuffle :: StdGen -> [a] -> [a]                                                                               
        do { let player = getPlayer playerNumber gameState;                                                           │shuffle gen = map snd . sortOn fst . zip (randoms gen :: [Int])                                               
             show (hand player);                                                                                      │                                                                                                              
             draw playerNumber <$> playCard playerNumber gameState }                                                  │exit :: IO ()                                                                                                 
      In an equation for ‘turn’:                                                                                      │exit = undefined                                                                                              
          turn playerNumber gameState                                                                                 │                                                                                                              
            = do { let player = ...;                                                                                  │winning :: GameState -> Bool                                                                                  
                   show (hand player);                                                                                │winning = undefined                                                                                           
                   draw playerNumber <$> playCard playerNumber gameState } 
```

@halogenandtoast could you please help me decipher the error? 🙏

### CODE REVIEW

**Comment by halogenandtoast** [comment]
@pauchan this function is in IO so each line has to be in IO as well. Keeping in mind that `do` is just syntax sugar, one of your problems is your use of `show` `show :: Show a => a -> String`

You can see show takes some generic value `a` that is in the typeclass `Show` and returns a `String`. A `String` isn't an `IO` action so this line is invalid. The error message is hard to parse, unless you remember that list is a monad. `Couldn't match type ‘[]’ with ‘IO’` means it wanted the monad IO, but got the monad List instead. So a simple solution would be to do:

```haskell
putStrLn $ show (hand player)
```

OR, even better is to use the `print` method which is simply `(putStrLn . show)`

```haskell
print $ hand player
```

**Comment by pauchan** [comment]
I've merged player hand rendering into master so this branch is not needed.
